tests/func/test_run_cache.py

```diff
@@ -0,0 +1,153 @@
+import os
+
+from dvc.dvcfile import PIPELINE_LOCK
+from dvc.utils import relpath
+
+
+def _recurse_count_files(path):
+    return len([os.path.join(r, f) for r, _, fs in os.walk(path) for f in fs])
+
+
+def test_push_pull(tmp_dir, dvc, erepo_dir, run_copy, setup_remote):
```

Comment:
The tests that are missing now are either low-level tests (which I don't really want to add), and more-higher level tests (eg: that covers CML like use-cases where you push to remote storage and remote branch, get contents to a different machine and try to repro or run).